app/repositories/migrations/schema6.js

```diff
@@ -0,0 +1,209 @@
+// schema version
+const schemaVersion = 6;
+
+// SCHEMAS
+const Coordinates = {
+  name: 'Coordinates',
+  properties: {
+    latitude: 'double',
+    longitude: 'double',
+    imageUrl: 'string?',
+    cdnImageUrl: 'string?',
+    currentloclat: 'double',
+    currentloclong: 'double',
+    isImageUploaded: 'bool?',
+    coordinateID: 'string?',
+  },
+};
+
+const Polygons = {
+  name: 'Polygons',
+  properties: {
+    isPolygonComplete: 'bool?',
+    coordinates: 'Coordinates[]',
+  },
+};
+
+const Species = {
+  name: 'Species',
+  properties: {
+    aliases: 'string',
+    treeCount: 'int',
+    id: 'string?',
+  },
+};
+
+const OfflineMaps = {
+  name: 'OfflineMaps',
+  primaryKey: 'name',
+  properties: {
+    areaName: 'string',
+    size: 'int',
+    name: 'string',
+  },
+};
+
+// used to record the sample trees
+const SampleTrees = {
+  name: 'SampleTrees',
+  properties: {
+    // stores the latitude of the tree
+    latitude: 'double',
+    // stores the longitude of the tree
+    longitude: 'double',
+    // stores the latitude of the device when location was marked
+    deviceLatitude: 'double',
+    // stores the longitude of the device when location was marked
+    deviceLongitude: 'double',
+    // stores accuracy of location when the location was marked
+    locationAccuracy: 'double?',
+    // URL of the image if picture was clicked
+    imageUrl: 'string?',
+    // CDN URL of the image if picture was clicked
+    cdnImageUrl: 'string?',
```

Comment:
Why should be store the CDN url in the database as it can be dynamically added to the image URL everywhere it's needed? This way we maybe store URL which do not work anymore if we change our CDN url.

Reply:
ImageUrl : Url of the Image stored in the file system
CDNImageUrl: Url of the image that we get in the response after uploading the image

There was only one property for both of these earlier. But, for displaying the images on the screen, I had to check first if that Path exists or not (in the Local File System), which was taking a little longer. That is why I thought of storing these in different properties. After uploading images, I'm making ImageUrl as an empty string. In this way, If the ImageUrl is false, it will fetch an image from the CDN using CDNImageUrl.

Reply:
Ok, thank you for the explanation. Then it make sense to have two URLs for distinction (otherwise one would have to have another boolean value user property to know the content type of the URL field). But do you store the full CDN url e.g. https://cdn.plant-for-the-planet.org/development/media/cache/coordinate/thumb/test.jpg or just the local part e.g. `/media/cache/coordinate/thumb/test.jpg` which I would suggest?

Reply:
No. I am not storing the entire Url, but just the name of an image.
i.e. test.jpg (From your given example)